integrade/tests/api/v1/test_cloud_accounts.py

```diff
@@ -0,0 +1,279 @@
+    :steps: 1) Create a user and authenticate with their password
+        2) Send a POST with the cloud account information to 'api/v1/account/'
+        3) Send a GET to 'api/v1/account/' to get a list of the cloud accounts
+        4) Attempt to create a duplicate and expect it to be rejected
+        5) Attempt to delete the account and expect to be rejected
+    :expectedresults: The server returns a 201 response with the information of
+        the created account, and the account cannot be duplicated or deleted.
+    """
+    user = create_user_account({
+        'email': '',
+        'password': gen_password(),
+        'username': uuid4(),
+    })
+    auth = get_auth(user)
+    client = api.Client(authenticate=False)
+    # make sure we are authenticated
+    client.get(urls.AUTH_ME, auth=auth)
+
+    cfg = config.get_config()
+    cloud_account = {
+        'account_arn': cfg['valid_role'],
+        'resourcetype': 'AwsAccount'
+    }
+    create_response = client.post(
+        urls.API_CLOUD_ACCOUNT,
+        payload=cloud_account,
+        auth=auth
+    )
+    assert create_response.status_code == 201
+
+    acct = create_response.json()
+
+    # get specific account
+    get_response = client.get(
+        urljoin(urls.API_CLOUD_ACCOUNT,
+                '{}/'.format(acct['id'])
+                ), auth=auth)
+    assert acct == get_response.json()
+
+    # list cloud accounts associated with this user
+    list_response = client.get(urls.API_CLOUD_ACCOUNT, auth=auth)
+    assert acct in list_response.json()['results']
+
+    # TODO need to try and update name, but
+    # feature is not delivered yet.
+
+    # TODO need to try and update arn, but
+    # feature is not delivered yet.
+
+    # assert we cannot create duplicate
+    client.response_handler = api.echo_handler
+    response = client.post(
+        urls.API_CLOUD_ACCOUNT,
+        payload=cloud_account,
+        auth=auth
+    )
+    assert response.status_code == 400
+
+    # attempt to delete the specific account
+    delete_response = client.delete(
+        urljoin(urls.API_CLOUD_ACCOUNT,
+                '{}/'.format(acct['id'])
+                ), auth=auth)
+    assert delete_response.status_code == 405
+
+
+@pytest.mark.serial_only
+@needs_valid_cloud_accounts
+def test_create_multiple_cloud_accounts(drop_account_data):
+    """Ensure cloud accounts can be registered to a user.
+
+    :id: d0174576-9b7c-48f7-8556-b560badf062d
+    :description: Ensure an user can register multiple cloud accounts as long
+        as they are ARNs to unique cloud accounts.
+    :steps: 1) Create a user and authenticate with their password
+        2) Send POSTS with each of the cloud account's information to
+            'api/v1/account/'
+        3) Send a GET to 'api/v1/account/' to get a list of the cloud accounts
+    :expectedresults: The server returns a 201 response with the information of
+        the created accounts.
+    """
+    user = create_user_account({
```

Comment:
This piece of code to handle user creation and getting its auth token is being repeated, what about creating a helper for this? I think this will be something that we may use many times